Source/Initialization/PlasmaInjector.cpp

```diff
@@ -72,10 +73,26 @@ CustomDensityProfile::CustomDensityProfile(const std::string& species_name)
 ParseDensityProfile::ParseDensityProfile(std::string parse_density_function)
     : _parse_density_function(parse_density_function)
 {
-    my_constants.ReadParameters();
-    parse_density_function = my_constants.replaceStringValue(parse_density_function);
     parser_density.define(parse_density_function);
     parser_density.registerVariables({"x","y","z"});
+
+    ParmParse pp("my_constants");
+    std::set<std::string> symbols = parser_density.symbols();
+    symbols.erase("x");
+    symbols.erase("y");
+    symbols.erase("z"); // after removing variables, we are left with constants
```

Comment:
Could you give a hint on what `symbols.erase` does? 
Can we use a constant which name includes `x`, like `x_min`, etc.?

Reply:
The code above is there to make sure all symbols are either constants are variables.  Because we have registered `x`, `y`, and `z` as variables, we erase them from the set of symbols.   Note that `symbols` is just local variable.  It will not change anything in the parser.  This is not to erase `x`, `y` and `z` from the parser.

I am going to add support for having underscore in names.